### ISSUE TITLE
Exempt orders adding 0.00% zip code entries

### ISSUE THREAD

**Issue by mjmarley** [taxjar/taxjar-woocommerce-plugin#156]
Hello,

We are noticing that exempt orders are erroneously adding zero percent entries in the Standard Tax Rates table. Shouldn't new entries only be added to this table when non-exempt orders are placed?

Please let us know ASAP if this is a bug, and if not, what would be the use case for the plugin behaving this way?

**Comment by dallendalton** [comment]
@mjmarley - Thanks for submitting this question. The TaxJar plugin utilizes and integrates with WooCommerce's default tax calculation, however there are some key differences.

WooCommerce's default tax calculation is based off a static tax table. When tax is calculated it looks at the cart, pulls out the shipping information (country, state, zip) and looks up the rate from the table to determine if it is taxed or not (at at what rate). 

However sales tax is much more complicated than just a simple lookup based on that shipping information. There are jurisdictions that use origin based taxation (tax is based on where the product ships from rather than where it is shipping to, areas with multiple different tax jurisdictions within a given zip code and city, jurisdictions that take into account the type, quantity and subtotal of the products to dynamically determine the rate, and a few other scenarios that a basic tax table can't account for.

In order to support all the scenarios and provide the most accurate rates TaxJar takes all the necessary cart data and performs a request to the TaxJar API for each unique cart. After getting the rates back from this request, the plugin then inserts them into the tax table to apply the rates to that single cart. When a new cart is created, even if it uses the same country, state and zip a new request to TaxJar will be sent and the rates specific to that cart will be applied. 

So in summary even though TaxJar is creating entries in the WooCommerce tax table, it is performing a unique request (and calculation) for each unique cart, and is only using the tax table to interface with the WooCommerce API in applying the rate to that unique cart. It has to do this even for fully and partially exempt orders, to ensure that real time rate TaxJar calculated is applied to the order.

**Comment by bhubbard** [comment]
Hi @dallendalton 
 
I work with @mjmarley, So at the moment we see how taxjar is saving to the table. In our specific store, we have many resellers, who have resell certificates. We mark all of them in TaxJar as exempt. We noticed however some of our clients **without** resell certificates are not getting charged tax, even when clearly in a state that charges tax such as California. 

We noticed how the exempt customers have their $0 tax being saved as a tax rate for their location. Can you confirm that WooCommerce is not using these $0 tax rates instead of a new correct tax amount provided by TaxJar?

I found the `create_or_update_tax_rate` function, I believe if we add a check to not add $0 tax this would prevent this issue.

```
			// Do not save empty rates.
			if ( empty( $rate ) || $rate === 0 ) {
				return;
			}
```

**Comment by dallendalton** [comment]
I have just confirmed on a fresh install of WordPress/WooCommerce/TaxJar that if an exempt customer places an order, and then another customer places an order with the exact same products and address (so the only difference is the customer ID associated with the cart) that the second customer is correctly taxed.

In order to troubleshoot your issue, please reach out to support@taxjar.com. You will also need to provide the email associated with your TaxJar account, a copy of your system status report, and order data that will allow us to track down the requests in question in our logs (usually the address on the order and the time it was created are enough). You can also reference this issue so that the support team has the necessary background (and so it will be brought to my attention in our systems). This will allow us to troubleshoot the issue without having to share any account/customer information publicly.

I'll leave this issue open and post the solution here as well once your ticket is resolved.

Thanks!
